### PR TITLE
Move ASAN and ONNX to Python 3.9 and 3.8

### PR DESCRIPTION
**Pull request by malfet**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #92787
* __->__ #92712
* #92711

As 3.7 is getting deprecated

### CODE REVIEW

**Review by weiwangmeta — APPROVED** [review]
LGTM!

**Review by kit1980 — APPROVED** [review]

**Review by seemethere — APPROVED** [review]

File: .circleci/docker/build.sh [hunk 1]
```diff
@@ -112,15 +112,15 @@ case "$image" in
     CONDA_CMAKE=yes
     ;;
   pytorch-linux-focal-py3-clang7-asan)
-    ANACONDA_PYTHON_VERSION=3.7
+    ANACONDA_PYTHON_VERSION=3.9
```
Comment (kit1980) [inline .circleci/docker/build.sh:115]: Why asan 3.9 but onnx 3.8?

Comment (malfet) [reply .circleci/docker/build.sh:115]: Because ONNX needs to run on minimally supported version, while for ASAN we don't care that much

**Pull request closed without merging**